packages/cli/src/gemini.tsx

```diff
@@ -159,17 +216,38 @@ export async function main() {
       if (settings.merged.selectedAuthType) {
         // Validate authentication here because the sandbox will interfere with the Oauth2 web redirect.
         try {
+          const authStart = performance.now();
           const err = validateAuthMethod(settings.merged.selectedAuthType);
           if (err) {
             throw new Error(err);
           }
           await config.refreshAuth(settings.merged.selectedAuthType);
+          const authEnd = performance.now();
+          const authDuration = authEnd - authStart;
+
+          // Record authentication performance if monitoring is active
+          if (isPerformanceMonitoringActive()) {
+            recordStartupPerformance(config, 'authentication', authDuration, {
+              auth_type: settings.merged.selectedAuthType,
+            });
+          }
         } catch (err) {
           console.error('Error authenticating:', err);
           process.exit(1);
         }
       }
+      const sandboxStart = performance.now();
```

Comment:
nit: all of these cases are a bit verbose with repeated logic to track a start time, end time and then compute the duration. An alternate more terse api could be something like
```
trackStartupPerformance(async () => {
   await start_sandbox(sandboxConfig, memoryArgs);
  },
  'sandbox_startup`
);
```
where the trackStartupPerformance method could have a no-op implementation if isPerformanceMonitoringActive()
is false.
As a bonus that would also make it easy to connect this to the performance.measure API in debug builds so that the times are also visible on the chrome devtools timeline.